### PR TITLE
Friendly Error Messages

### PR DESCRIPTION
**Pull request by njjc**
# Critical Changes

# Changes
- Validation rules and required fields now display more user friendly error messages.

# Issues Closed

### CODE REVIEW

File: src/classes/ERR_ExceptionHandler.cls [hunk 1]
```diff
@@ -0,0 +1,16 @@
+public with sharing class ERR_ExceptionHandler {
+    private static final Map<String, String> exceptionMessages = new Map<String, String>{
+        'FIELD_CUSTOM_VALIDATION_EXCEPTION' => 'A validation rule is preventing the record from saving: ',
```
Comment (njjc) [inline src/classes/ERR_ExceptionHandler.cls]: The values of this map would be custom labels in a final implementation.

File: src/classes/ERR_ExceptionHandler.cls [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        'REQUIRED_FIELD_MISSING' => 'A required field is missing: '
+    };
+
+    public static void beautifyException(Exception ex) {
+        String message = ex.getMessage();
+        for (String status : exceptionMessages.keySet()) {
+            if (message.contains(status)) {
+                message = exceptionMessages.get(status) + message.substringBetween(status, ':');
```
Comment (davidhabib) [inline src/classes/ERR_ExceptionHandler.cls]: so does this end up still preserving the name of the field?  I think it does, but wanted to check.

File: src/classes/TDTM_TriggerHandler.cls [hunk 3]
```diff
@@ -107,6 +124,8 @@ public class TDTM_TriggerHandler {
                 UTIL_Debug.debugWithInfo('****Finished executing: ' + classToCall.get('Class__c'));
             }
 
+            depthCounter++;
```
Comment (davidhabib) [inline src/classes/TDTM_TriggerHandler.cls]: when does this get decremented?

**Comment by njjc** [comment]
This isn't quite working yet, please hold off on comments until I can fix it.

**Comment by ceiroa** [comment]
Please keep me posted on when this is ready for review. I'd like to get my head wrapped around this. Thanks!

**Comment by njjc** [comment]
@jjbennett this is ready for review. I couldn't think of a way to get a handled exception through a trigger in a test, so I just wrote unit tests and not end to end tests. Maybe we can brainstorm a way to do that together.

**Pull request merged**